src/Microsoft.TestPlatform.Common/Utilities/StringExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System;
+
+#if NETFRAMEWORK || WINDOWS_UAP || (NETSTANDARD && !NETSTANDARD2_1_OR_GREATER)
+internal static class StringExtensions
```

Comment:
I think we should consider this PR to be added to blame ignore. Make sure you put this in a separate commit, and don't squash.

Reply:
Are we including this into every project? We could also leverage it to replace `string.IsNullOrWhitespace(someString)` and some of the other annoyingly difficult things we do with strings and other types. If we don't include it everywhere we might want to move this to a shared project and import that here and explicitly to other projects so we avoid adding an extra dll we need to manage.

Reply:
About the new project, I guess you are thinking about some shared project rather than a class library project? Could we rename `Microsoft.TestPlatform.Execution.Shared` into `Microsoft.TestPlatform.Shared` and make it referenced by all projects? Or do you prefer a separate project?

Reply:
@nohwnd Merging without squashing is not possible as per our current repo config.

Reply:
I can cherry-pick specific commit (or copy files) and create a separate PR when we agree on the scope.

Reply:
@Haplois I thought so, we can change it for this PR or cherry pick the commit as mentioned above.